rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/security/ResourceAuthenticationFilter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Siemens AG, 2018. Part of the SW360 Portal Project.
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.sw360.rest.resourceserver.security;
+
+import org.apache.log4j.Logger;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Profile("!SECURITY_MOCK")
+public class ResourceAuthenticationFilter implements Filter {
+
+    private static final Logger log = Logger.getLogger(ResourceAuthenticationFilter.class);
+    private static final String AUTHENTICATION_TOKEN_PARAMETER = "authorization";
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+        SecurityContext context = SecurityContextHolder.getContext();
+
+        if (context.getAuthentication() != null && context.getAuthentication().isAuthenticated()) {
+            log.debug("Already authenticated");
+        } else {
+            Map<String, String> headers = Collections.list(((HttpServletRequest) request).getHeaderNames()).stream()
+                    .collect(Collectors.toMap(h -> h, ((HttpServletRequest) request)::getHeader));
+            if (!headers.isEmpty() && headers.containsKey(AUTHENTICATION_TOKEN_PARAMETER)) {
+                String token = headers.get(AUTHENTICATION_TOKEN_PARAMETER);
+                token = token.replaceAll("(?i)token", "").trim();
+                if (token != null) {
+                    Authentication auth = new TokenAuthentication(token);
+                    SecurityContextHolder.getContext().setAuthentication(auth);
+                }
+            }
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    class TokenAuthentication implements Authentication {
+        private String token;
+
+        private TokenAuthentication(String token) {
+            this.token = token;
+        }
+
+        @Override
+        public Collection<? extends GrantedAuthority> getAuthorities() {
+            return new ArrayList<>(0);
```

Comment:
Why `ArrayList<>(0)` instead of `ArrayList<>()`?

Reply:
its an unused override method of `Authentication`. But I changed the logic to  `return new ArrayList<>();`